msupply-foundation/open-msupply
# 3119 omsupply central sync steps

Fixes #3119

# 👩🏻‍💻 What does this PR do? 
 

Exposes central v6 steps on the sync page, and maps v6 errors:
![Screenshot 2024-04-19 at 9 09 20 AM](https://github.com/msupply-foundation/open-msupply/assets/55115239/c046c4c6-bb92-43d3-a632-fa1afb0d4049)

# 🧪 How has/should this change been tested? 


Uses #3635 as a base, you'll need to use latest develop of mSupply to include the [configuration changes](https://github.com/msupply-foundation/msupply/pull/14581) (See those PRs for setup)


- In mSupply, don't add (or remove) the `This site url` for your omSupply central server
- Start remote sync
- You should see an error in the Sync UI (V6 central server not configured)
- Add the site URL
- Sync page should now show all 6 sync steps
- v6 errors also mapped:
  - stop oms central server - UI shows Unable to connect to server rather than unknown sync error...
  - (how to test parsing error?)
  - should more errors be mapped to new codes?


## 💌 Any notes for the reviewer?


The totals beneath the push steps are for all changelogs still to be pushed (not filtered by v5/v6) - that filtering is done by the translators, and it's nice to leave that logic in one place.

It was discussed about changing/adding to show total SENT though - this is possibly quite helpful from a devving/troubleshooting standpoint, to know how many records were sent to which central server... not sure how we would display though (i.e. `processed 5 / 10 records, sent 2 to central v6`...)

## 📃 Documentation


- [ ] Update screenshots for synchronisation docs
